src/components/header/buttons/PeersButton.js

```diff
@@ -7,28 +7,30 @@ export default class PeersButton extends Component {
   constructor (props) {
     super(props)
 
-    const initialState = { peers: {}, dropdownOpen: false }
-
-    if (props.peerGroup) {
-      initialState.peers = props.peerGroup.all()
-      props.peerGroup.on('change', this.onPeersChange)
+    const initialState = {
+      peers: {},
+      dropdownOpen: false,
+      alias: props.alias || ''
     }
 
     this.state = initialState
 
     this.onPeersChange = this.onPeersChange.bind(this)
     this.onDropdownClick = this.onDropdownClick.bind(this)
     this.onDropdownDismiss = this.onDropdownDismiss.bind(this)
+    this.onAliasChange = this.onAliasChange.bind(this)
+    this.onSaveAlias = this.onSaveAlias.bind(this)
+
+    if (props.peerGroup) {
+      initialState.peers = props.peerGroup.all()
```

Comment:
This would read more naturally if we pulled the 
```js
this.state = initialState
```
line after setting the peer group on the initialState object.